ostelco-ios-client/Coordinators/EKYC/SingaporeEKYCCoordinator.swift

```diff
@@ -0,0 +1,222 @@
+        case stepsForNRIC
+        case enterNRIC
+        case ekycRejected
+        case success(region: RegionResponse)
+        case waitingForVerification
+    }
+    
+    let navigationController: UINavigationController
+    
+    weak var delegate: EKYCCoordinatorDelegate?
+    var jumioCoordinator: JumioCoordinator?
+    var singPassCoordinator: SingPassCoordinator?
+    let country = Country("sg")
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+    }
+    
+    func determineAndNavigateDestination(from region: RegionResponse?, animated: Bool) {
+        let destination = self.determineDestination(from: region)
+        self.navigate(to: destination, animated: animated)
+    }
+    
+    func determineDestination(from region: RegionResponse?) -> SingaporeEKYCCoordinator.Destination {
+        guard let region = region else {
+            return .landing
+        }
+        
+        switch region.status {
+        case .APPROVED:
+            return .success(region: region)
+        case .REJECTED:
+            return .ekycRejected
+        case .PENDING:
+            guard
+                let jumio = region.kycStatusMap.JUMIO,
+                let addressAndPhoneNumber = region.kycStatusMap.ADDRESS_AND_PHONE_NUMBER,
+                let nricFin = region.kycStatusMap.NRIC_FIN,
+                let myInfo = region.kycStatusMap.MY_INFO else {
+                    return .goBackAndChooseCountry
+            }
+            
+            switch (jumio, addressAndPhoneNumber, nricFin, myInfo) {
+            case (.PENDING, .PENDING, .PENDING, .PENDING):
+                // User hasn't selected anything yet
+                return .selectVerificationMethod
+            case (.REJECTED, _, _, _):
+                // If jumio is rejected, *everything* is rejected.
+                return .ekycRejected
+            case (_, _, _, .REJECTED):
+                // If myInfo is rejected, *everything* is rejected
+                return .ekycRejected
+            case (.APPROVED, .APPROVED, .APPROVED, _):
+                // Jumio + addresss + NRCFIN = Yay!
+                return .success(region: region)
+            case (_, .APPROVED, _, .APPROVED):
+                // MyInfo + address = yay!
+                return .success(region: region)
+            case (_, .PENDING, _, .APPROVED):
+                // My info also needs an address submitted, but we've probably lost
+                // what we need to access their singpass address. Make the user enter it.
+                return .enterAddress
+            case (.PENDING, .PENDING, .APPROVED, _):
+                // The user has an approved NRIC. Now we need Jumio verification and an address. Start with jumio.
+                return .jumio
+            case (.PENDING, .APPROVED, .APPROVED, _):
+                // The user has an approved NRIC and address, now we need Jumio verification.
+                return .jumio
```

Comment:
With this code path, Will go through address after Jumio (Which is already approved)?